src/main/java/hw21/Product.java

```diff
@@ -0,0 +1,82 @@
+package hw21;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+import static hw21.ProductType.BOOK;
+
+public class Product {
+    private final int id;
+    private final ProductType productType;
+    private double price;
+    private final boolean discount;
+    private final LocalDate supplyDate;
+
+    public Product(int id, ProductType productType, double price, boolean discount, LocalDate supplyDate) {
+        this.id = id;
+        this.productType = productType;
+        this.price = price;
+        this.discount = discount;
+        this.supplyDate = supplyDate;
+    }
+
+    public boolean isBook() {
+        return getType() == BOOK;
+    }
+
+    public boolean isCurrentYearSupply() {
+        LocalDate now = LocalDate.now();
+        return getSupplyDate().getYear() == now.getYear();
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public ProductType getType() {
+        return productType;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public void setPrice(double price) {
+        if (isDiscount()) {
+            this.price = price;
+        }
+    }
+
+    public boolean isDiscount() {
```

Comment:
Would be better to give it the name `hasDiscount`, so it is easier to understand the context of the method.